lib/plugins/fs-routes/router-template.js

```diff
@@ -0,0 +1,127 @@
+const path = require('path')
+const fs = require('fs-extra')
+const sortBy = require('lodash.sortby')
+
+module.exports = async (api, pagesDir) => {
+  const routes = await collectRoutes({
+    dir: api.resolveBaseDir(pagesDir),
+    componentPrefix: path.join('#base/', pagesDir),
+    parentPath: '/'
+  })
+
+  if (routes.length === 0) {
+    return `
+    export default function () {
+      // No routes found in ${pagesDir}
+    }`
+  }
+
+  function renderRoutes(routes) {
+    return routes
+      .map(
+        route => `
+    {
+      path: ${JSON.stringify(route.path)},
+      component: () => import(${JSON.stringify(route.component)}),
+      ${route.children ? `children: [${renderRoutes(route.children)}]` : ``}
+    }`
+      )
+      .join(',')
+  }
+
+  return `
+  import Router from 'vue-router'
+
+  export default function () {
+    return new Router({
+      mode: 'history',
+      routes: [
+        ${renderRoutes(routes)}
+      ]
+    })
+  }`
+}
+
+// index.vue -> /
+// about.vue -> /about
+// user.vue -> /user
+// user/index.vue -> /user, child ''
+// user/friends.vue -> /user, child 'friends'
+// catalog/index.vue -> /catalog
+// catalog/specials.vue -> /catalog/specials
+
+async function collectRoutes({ dir, componentPrefix, parentPath }) {
```

Comment:
maybe use `fast-glob` to simplify this, and `chokidar` to add watch functionality.

Reply:
Agreed with `chokidar`, but I don't think `fast-glob` will be simpler. The reasons being:
* non-uniform directory recursion (a directory may either generate child routes with cleared path prefix, or extend path prefix and append to the parent routes)
* per-directory sorting